internal/scorecard/alpha/scorecard.go

```diff
@@ -23,70 +23,69 @@ import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
 	"github.com/operator-framework/operator-sdk/version"
 )
 
+type TestRunner interface {
+	Initialize(context.Context) (string, error)
+	RunTest(context.Context, string, Test) (*v1alpha2.ScorecardTestResult, error)
+	Cleanup(context.Context, string) error
+}
+
 type Scorecard struct {
-	Config              Config
-	Selector            labels.Selector
-	BundlePath          string
-	WaitTime            time.Duration
-	Namespace           string
-	ServiceAccount      string
-	bundleConfigMapName string
-	Client              kubernetes.Interface
-	SkipCleanup         bool
+	Config      Config
+	Selector    labels.Selector
+	TestRunner  TestRunner
+	SkipCleanup bool
 }
 
-// RunTests executes the scorecard tests as configured
-func (o Scorecard) RunTests() (testOutput v1alpha2.ScorecardOutput, err error) {
-	tests := o.selectTests()
-	if len(tests) == 0 {
-		testOutput.Results = make([]v1alpha2.ScorecardTestResult, 0)
-		return testOutput, err
-	}
+type PodTestRunner struct {
+	Namespace      string
+	ServiceAccount string
+	BundlePath     string
+	Client         kubernetes.Interface
+	ConfigMapName  string
+}
+
+type FakeTestRunner struct {
+	TestResult *v1alpha2.ScorecardTestResult
+	Error      error
+}
 
-	ctx, cancel := context.WithTimeout(context.Background(), o.WaitTime)
-	defer cancel()
+// RunTests executes the scorecard tests as configured
+func (o Scorecard) RunTests(ctx context.Context) (testOutput v1alpha2.ScorecardOutput, err error) {
 
-	bundleData, err := o.getBundleData()
+	configMapName, err := o.TestRunner.Initialize(ctx)
```

Comment:
I don't think we want the generic `RunTests` function to have any knowledge that a configmap is being created.  Since the configMapName is being returned from and consumed by TestRunner commands only, we should be able to add `configMapName` as an unexported field on the PodTestRunner struct.

We can set it during initialize and then use it during runtest and cleanup.

Reply:
updated in latest commit.